mesa/examples/advanced/wolf_sheep/agents.py

```diff
@@ -62,6 +63,27 @@ def feed(self):
             self.energy += self.energy_from_food
             grass_patch.fully_grown = False
 
+    def move(self):
+        """Move towards a cell where there isn't a wolf, and preferably with grown grass."""
+        cells_without_wolves = self.cell.neighborhood.select(
+            lambda cell: not any(isinstance(obj, Wolf) for obj in cell.agents)
```

Comment:
@quaquel I think this eats some performance. Ideas on how to do it faster?

Reply:
Tricky. 

1. you could make it a normal for loop and jump out the moment you see a wolf. This will shave of a little bit of performance
2. Use a property layer of booleans and modify the wolf to set this property to True/False in their move method. Now you can express the entire movement in a vectorized manner.

Reply:
Agree it's tricky, wrote it up for now in #2565.